### PR TITLE
10.2.0 gt showquestions

### PR DESCRIPTION
**Pull request by adamcrow64**
Button Events now showing! but alyson template needs fixing.

### CODE REVIEW

**Review by weslyvo81 — APPROVED** [review]
pcm form edit and pcm events are added. Looks good

File: kogitoq/kogito-common/src/main/java/life/genny/kogito/common/core/Dispatch.java [hunk 1]
```diff
@@ -111,6 +111,16 @@ public QBulkMessage build(ProcessData processData, PCM pcm) {
 		if (buttonEvents != null) {
 			Ask eventsAsk = createButtonEvents(buttonEvents, sourceCode, targetCode);
 			msg.add(eventsAsk);
+
+			PCM eventsPCM = beUtils.getPCM("PCM_EVENTS");
+
+			// Now update the PCM to point the last location to the PCM_EVENTS
+			// add a LOC2 to the PCM if it doesn't exist
+			if (pcm.getLocation(2) == null) {
+				pcm.addStringAttribute("PRI_LOC2", "LOC2", PCM.PCM_EVENTS);
```
Comment (jasperRob) [inline kogitoq/kogito-common/src/main/java/life/genny/kogito/common/core/Dispatch.java:131]: use PCM method `setLocation()` instead

File: kogitoq/kogito-common/src/main/java/life/genny/kogito/common/core/Dispatch.java [hunk 2]
```diff
@@ -111,6 +111,16 @@ public QBulkMessage build(ProcessData processData, PCM pcm) {
 		if (buttonEvents != null) {
 			Ask eventsAsk = createButtonEvents(buttonEvents, sourceCode, targetCode);
 			msg.add(eventsAsk);
+
+			PCM eventsPCM = beUtils.getPCM("PCM_EVENTS");
```
Comment (jasperRob) [inline kogitoq/kogito-common/src/main/java/life/genny/kogito/common/core/Dispatch.java:124]: use PCM static constant

File: kogitoq/kogito-common/src/main/java/life/genny/kogito/common/core/Dispatch.java [hunk 3]
```diff
@@ -251,6 +261,10 @@ public void traversePCM(PCM pcm, BaseEntity source, BaseEntity target,
 			QBulkMessage msg, ProcessData processData) {
 
 		log.info("Traversing " + pcm.getCode());
+		if ("PCM_FORM_EDIT".equals(pcm.getCode())) { // TODO, this could be better
```
Comment (jasperRob) [inline kogitoq/kogito-common/src/main/java/life/genny/kogito/common/core/Dispatch.java:278]: use PCM static constant

File: kogitoq/kogito-common/src/main/java/life/genny/kogito/common/service/TaskService.java [hunk 4]
```diff
@@ -157,6 +157,12 @@ public ProcessData dispatchFull(String sourceCode, String targetCode, String que
 			throw new NullParameterException("pcmCode");
 		if (buttonEvents == null)
 			throw new NullParameterException("buttonEvents");
+		if (parent == null) {
```
Comment (jasperRob) [inline kogitoq/kogito-common/src/main/java/life/genny/kogito/common/service/TaskService.java:163]: Looks like this is unnecessarily duplicated

**Review by jasperRob — COMMENTED** [review]
Address these comments

**Pull request closed without merging**